assets/rpip-52/rewards-file-spec.md

```diff
@@ -0,0 +1,102 @@
+                                              # Sorted ascending by network
+    node_rewards: List[NodeReward, 9223372036854775807] # Per-node rewards
+                                                        # Sorted in ascending unsigned numerical order by address
+
+class TotalRewards(Container):
+    protocol_dao_rpl: uint256 # Amount of RPL sent to the pDAO (in wei)
+    total_collateral_rpl: uint256 # Amount of RPL sent to Node Operators (in wei)
+    total_oracle_dao_rpl: uint256 # Amount of RPL sent to oDAO members (in wei)
+    total_smoothing_pool_eth: uint256 # Amount of ETH in the Smoothing Pool (in wei)
+    pool_staker_smoothing_pool_eth: uint256 # Amount of ETH sent to the rETH contract (in wei)
+    node_operator_smoothing_pool_eth: uint256 # Amount of ETH send to Node Operators in the Smoothing Pool (in wei)
+    total_node_weight: uint256 # Total Node Weight as defined by RPIP-30 (in wei)
+
+class NetworkReward(Container):
+    network: uint64 # The L1 or L2 id that this object describes rewards for
+                    # Currently, only mainnet is supported, with id of 0
+    collateral_rpl: uint256 # Amount of RPL for Node Operators sent to this network (in wei)
+    oracle_dao_rpl: uint256 # Amount of RPL for oDAO Operators sent to this network (in wei)
+    smoothing_pool_eth: uint256 # Amount of ETH for Node Operators sent to this network (in wei)
+
+class NodeReward(Container):
+    address: Bytes20 # Address of the node that this object describes rewards for
+    network: uint64 # The L1 or L2 id that the node will claim on
+                    # Corresponds to a NetworkReward.network in the File.network_rewards list
+                    # 0 means L1
+    collateral_rpl: uint256 # RPL staking rewards earned by this node operator (in wei)
+    oracle_dao_rpl: uint256 # RPL rewards earned by this oDAO member (in wei)
+                            # If the node is not in the oDAO the expected value is 0
+    smoothing_pool_eth: uint256 # Smoothing Pool ETH earned by this node operator (in wei)
+                                # If the node was not in the smoothing pool during the interval,
+                                # the expected value is 0
+```
+
+# Naming
+
+The file SHALL be serialized with the binary representation of the above specified format and named according to convention.
+
+The file SHALL be named `rp-rewards-{NETWORK}-{INTERVAL}.ssz` with the following substitutions:
+
+  * Substitute the name of the network from the below table for {NETWORK}
+  * Substitute the number of the interval, in decimal with no leading zeroes, for {INTERVAL}
+
+## Networks
+| Network | {NETWORK} |
+| ------- | --------- |
+| Mainnet | mainnet   |
+| Holešky | holesky   |
+| devnet  | devnet    |
+
+# CID Calculation
+
+The Oracle DAO SHALL calculate an IPFS CID when submitting the Merkle Root for consensus to the rocketRewardsContract.
+
+The IPFS CID shall be computed with:
```

Comment:
Might we need to also import these IPFS specs into our assets? Or link to a specific commit? (Do they change much?)

Reply:
I don't think they change at all

Reply:
I still think we might need to link to a specific commit to avoid ambiguity.